### PR TITLE
enh(tests): Add visual tests for status resources filter

### PR DESCRIPTION
**Pull request by Med1233**
## Description

Rework Jest RTL tests to Cypress component testing 

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+
+      cy.findByPlaceholderText(labelSearch).clear();
+      cy.findByPlaceholderText(labelSearch).type(fieldSearchValue);
+      cy.findByLabelText(labelSearchOptions).click();
+      cy.findByText(labelSearch).click();
+      cy.waitForRequest(`@request/${searchableField}`);
+
+      cy.matchImageSnapshot();
+    });
+  });
```
Comment (bdauria) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:205]:
```suggestion
  });
  
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+    customFilter.forEach(([filterGroup, criterias]) => {
+      const endpoint = getListingEndpoint(criterias as EndpointParams);
+
+      cy.interceptAPIRequest({
+        alias: `request/${filterGroup}`,
+        method: Method.GET,
+        path: Ramda.replace('./api/latest/monitoring', '**', endpoint)
+      });
+    });
+  });
```
Comment (bdauria) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:311]:
```suggestion
  });
  
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,345 @@
+/* eslint-disable @typescript-eslint/no-unused-expressions */
```
Comment (bdauria) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:1]: what's unused? can't we get rid of it?

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+/* eslint-disable react/jsx-no-constructed-context-values */
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:2]: I think you can get rid of this too

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+  getListingEndpoint,
+  searchableFields,
+  EndpointParams
+} from '../testUtils';
+
+import useFilter from './useFilter';
+
+import Filter from '.';
+
+const fakeData = {
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:40]:
```suggestion
const lemptyListData = {
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,345 @@
...
+  userData.result.current.locale = 'en_US';
+});
+
+describe('Filter', () => {
+  beforeEach(() => {
+    cy.interceptAPIRequest({
+      alias: 'filterRequest',
+      method: Method.GET,
+      path: '**/events-view*',
+      response: fakeData
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:149]:
```suggestion
      response: emptyListData
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,345 @@
...
+    });
+
+    const DefaultEndpoint = getListingEndpoint({});
+
+    cy.interceptAPIRequest({
+      alias: `defaultRequest`,
+      method: Method.GET,
+      path: Ramda.replace('./api/latest/monitoring', '**', DefaultEndpoint),
+      response: fakeData
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:158]:
```suggestion
      response: emptyListData
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,345 @@
...
+      states: [],
+      statusTypes: [],
+      statuses: []
+    });
+
+    cy.interceptAPIRequest({
+      alias: `getListRequest`,
+      method: Method.GET,
+      path: Ramda.replace('./api/latest/monitoring', '**', endpoint),
+      response: fakeData
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:221]:
```suggestion
      response: emptyListData
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,345 @@
...
+    });
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:222]: You already define this request interception in the beforeEach

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,345 @@
...
+  });
+});
+
+describe('Custom filter', () => {
+  beforeEach(() => {
+    cy.interceptAPIRequest({
+      alias: 'filterRequest',
+      method: Method.GET,
+      path: '**/events-view*',
+      response: fakeData
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:254]:
```suggestion
      response: defaultRequest
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 11]
```diff
@@ -0,0 +1,345 @@
...
+    });
+
+    cy.interceptAPIRequest({
+      alias: 'hostgroups',
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:258]:
```suggestion
      alias: 'hostGroupsRequest',
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 12]
```diff
@@ -0,0 +1,345 @@
...
+      method: Method.GET,
+      path: '**/hostgroups?*',
+      response: {
+        meta: { limit: 10, page: 1, search: {}, sort_by: {}, total: 1 },
+        result: [linuxServersHostGroup]
+      }
+    });
+
+    cy.interceptAPIRequest({
+      alias: 'dataToListingTable',
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:268]:
```suggestion
      alias: 'serviceGroupsRequest',
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 13]
```diff
@@ -0,0 +1,345 @@
...
+    )} filter is set`, () => {
+      cy.findByText(labelUnhandledProblems).click();
+
+      cy.findByText(filterGroup).click();
+
+      cy.waitForRequest(`@request/${filterGroup}`);
+    });
+  });
+
+  filterParams.forEach(([criteriaName, optionToSelect, _]) => {
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:324]:
```suggestion
  filterParams.forEach(([criteriaName, optionToSelect]) => {
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 14]
```diff
@@ -0,0 +1,345 @@
...
+  [
+    labelServiceGroup,
+    webAccessServiceGroup.name,
+    {
+      serviceGroups: [webAccessServiceGroup.name]
+    }
+  ]
+];
+
+const customFilter = [
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:96]:
```suggestion
const customFilters = [
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 15]
```diff
@@ -0,0 +1,345 @@
...
+      });
+    });
+
+    cy.mount({
+      Component: <FilterWithProvider />
+    });
+
+    cy.viewport(1200, 1000);
+
+    customFilter.forEach(([filterGroup, criterias]) => {
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:302]:
```suggestion
    customFilters.forEach(([filterGroup, criterias]) => {
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 16]
```diff
@@ -0,0 +1,345 @@
...
+      const endpoint = getListingEndpoint(criterias as EndpointParams);
+
+      cy.interceptAPIRequest({
+        alias: `request/${filterGroup}`,
+        method: Method.GET,
+        path: Ramda.replace('./api/latest/monitoring', '**', endpoint)
+      });
+    });
+  });
+  customFilter.forEach(([filterGroup, criterias]) => {
```
Comment (Thebarda) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:312]:
```suggestion
  customFilters.forEach(([filterGroup, criterias]) => {
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 17]
```diff
@@ -0,0 +1,345 @@
...
+    searchableFields.forEach((searchableField) => {
+      const search = 'foobar';
+      const fieldSearchValue = `${searchableField}:${search}`;
+      const endpoint = getListingEndpoint({
+        resourceTypes: [],
+        search: fieldSearchValue,
+        states: [],
+        statusTypes: [],
+        statuses: []
+      });
```
Comment (SoldierBoy31) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:176]:
```suggestion
      });
      
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 18]
```diff
@@ -0,0 +1,345 @@
...
+    it(`executes a listing request with an "$and" search param containing ${searchableField} when ${searchableField} is typed in the search field`, () => {
+      cy.waitForRequest('@filterRequest');
+      const search = 'foobar';
+      const fieldSearchValue = `${searchableField}:${search}`;
+
+      cy.findByPlaceholderText(labelSearch).clear();
+      cy.findByPlaceholderText(labelSearch).type(fieldSearchValue);
+      cy.findByLabelText(labelSearchOptions).click();
+      cy.findByText(labelSearch).click();
+      cy.waitForRequest(`@request/${searchableField}`);
```
Comment (SoldierBoy31) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:201]:
```suggestion
      cy.waitForRequest(`@request/${searchableField}`);
      
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx [hunk 19]
```diff
@@ -0,0 +1,345 @@
...
+
+  it('executes a listing request with "Unhandled problems" filter by default', () => {
+    cy.waitForRequest('@defaultRequest');
+
+    cy.matchImageSnapshot();
+  });
+
+  searchableFields.forEach((searchableField) => {
+    it(`executes a listing request with an "$and" search param containing ${searchableField} when ${searchableField} is typed in the search field`, () => {
+      cy.waitForRequest('@filterRequest');
```
Comment (SoldierBoy31) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:193]:
```suggestion
      cy.waitForRequest('@filterRequest');
      
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx (hunk 1 above)
Comment (SoldierBoy31) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:205]:
```suggestion
  });
  
```

File: centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx (hunk 9 above)
Comment (SoldierBoy31) [inline centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx:222]:
```suggestion
    });
    
```

**Review by Thebarda — COMMENTED** [review]
Why did you remove the whole snapshot folder?

**Comment by bdauria** [comment]
@Med1233 this diff file should be removed

**Comment by Thebarda** [comment]
btw, rename Fitter.Storage.cypress.spec.tsx to FilterStorage.cypress.spec.tsx

**Comment by Med1233** [comment]
@bdauria  I can't see the image! could you please upload it again

**Comment by bdauria** [comment]
> @bdauria I can't see the image! could you please upload it again

image

**Comment by Thebarda** [comment]
@Med1233 You can remove all __diff_output__ folders and git ignore them

**Review by bdauria — APPROVED** [review]

**Review by Thebarda — APPROVED** [review]

**Pull request merged**